content/200-orm/200-prisma-client/100-queries/063-excluding-fields.mdx

```diff
@@ -4,15 +4,163 @@ metaTitle: 'Excluding fields'
-
 By default Prisma Client returns all fields from a model. You can use `select` to narrow the result set, but that can be unwieldy if you have a large model and you only want to exclude one or two fields.
 
-Prisma Client doesn't have a native way of excluding fields yet, but it's easy to create a function that you can use to exclude certain fields in a type-safe way.
+:::info
+
+As of Prisma ORM 5.16.0, excluding fields globally and locally is supported via the `omitApi` Preview feature.
+
+:::
+
+## Excluding a field globally using `omit`
+
+The following is a type-safe way to exclude a field globally using the [`omitApi` Preview feature](/orm/reference/preview-features):
+
+<TabbedContent code>
+
+<TabItem value="Schema">
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["omitApi"]
+}
+
+model User {
+  id        Int      @id @default(autoincrement())
+  firstName String
+  lastName  String
+  email     String   @unique
+  password  String
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+}
+```
+
+</TabItem>
+
+<TabItem value="Code">
 
-</TopBlock>
+```tsx
+const prisma = new PrismaClient({
+  omit: {
+    user: {
+      password: true
+    }
+  }
+})
+
+// The password field is excluded in all queries, including this one
+const user = await prisma.user.findUnique({ where: { id: 1 } })
+```
+
+</TabItem>
+
+</TabbedContent>
+
+## Excluding a field locally using `omit`
+
+The following is a type-safe way to exclude a field locally using the `omitApi` Preview feature:
 
-## Excluding the password field
+<TabbedContent code>
+
+<TabItem value="Schema">
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["omitApi"]
+}
+
+model User {
+  id        Int      @id @default(autoincrement())
+  firstName String
+  lastName  String
+  email     String   @unique
+  password  String
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+}
+```
+
+</TabItem>
+
+<TabItem value="Code">
+
+```tsx
+const prisma = new PrismaClient()
+
+// The password field is excluded only in this query
+const user = await prisma.user.findUnique({
+  omit: {
+    password: true
+  },
+  where: { 
+    id: 1 
+  } 
+})
+```
+
+</TabItem>
+
+</TabbedContent>
+
+## How to omit multiple fields
+
+Omitting multiple fields works the same as selecting multiple fields: add multiple key-value pairs to the omit option.
+Using the same schema as before, you could omit password and email with the following:
+
+```tsx
+const prisma = new PrismaClient()
+
+// password and email are excluded
+const user = await prisma.user.findUnique({
+  omit: {
+    email: true,
+    password: true,
+  },
+  where: {
+    id: 1,
+  },
+})
+```
+
+Multiple fields can be omitted locally and globally.
+
+## How to select a previously omitted field
+
+If you [omit a field globally](#excluding-a-field-globally-using-omit), you can select it by setting `omit` to `false` in any query.
```

Comment:
Global `omit` also does not apply to explicit `select`.
So, `{ select: { password: true } }` will also include the password.